src/Utils/Image.php

```diff
@@ -514,6 +514,8 @@ public function save($file = NULL, $quality = NULL, $type = NULL)
 					break;
 				case 'gif':
 					$type = self::GIF;
+				default:
+					throw new Nette\InvalidArgumentException('Unsupported file extension \'' . pathinfo($file, PATHINFO_EXTENSION) . '\'.');
```

Comment:
You can also rewrite this with double quotes so you don't have to escape, also you can reuse pathinfo output from the switch above?
```php
switch (strtolower($ext = pathinfo($file, PATHINFO_EXTENSION))) {
```